### PR TITLE
refactor(health): consolidate all backend health classification into pkg/health

### PR DESCRIPTION
**Pull request by nadaverell**
## Summary

Radar computed Kubernetes resource health in ~8 independent places across three Go vocabularies and four divergent rank maps — they drifted, and the drift caused real bugs (a crashlooping pod showing **green** in the topology graph while the table showed it red; an `unknown` tier that ranked *below* healthy in one rank map and *above* it in another). This PR makes **`pkg/health` the single backend classifier** and deletes every duplicate.

This is **PR 1 of 2**. A follow-up handles the frontend (backend emits the health field on the wire; the TS classifiers become a fallback; `neutral` renders as its own tier).

## What's deleted (the consolidation)

- **`internal/k8s/health.go`** — the entire file, including the shim layer. Every caller (`detect`, `detect_node`, `dashboard`, `mcp`, `applications`, `packages`, `timeline`, `topology`, `resourcecontext`) now calls `pkg/health` directly.
- **9 duplicate classifier functions** (`jobHealth`, `cronJobHealth`, `deploymentHealth`, `daemonsetHealth`, `statefulsetHealth`, `worstAppHealth`, `appHealthRank`, `podHealth`, `replicasHealth`).
- **3 of the 4 rank maps** — `timeline` and `packages` worst-of now both delegate to `health.Rank`.
- the phase-blind topology pod/PVC classifiers; the inline topology CronJob + replica-ratio workload classifiers.

## What's added

- **`pkg/health`** — a pure leaf package: `Level` (healthy/neutral/degraded/unhealthy/unknown, 5-tier, internal) + one `Rank`/`WorseOf`; `Pod`/`Workload`/`Node` → `Verdict{Level, Reason, Message}`; clock-injected. `Workload` unifies the four replica/job/pvc copies.
- **The wire vocab stays 4-value** (healthy/degraded/unhealthy/unknown), same as `main`. `neutral` is internal to `health.Level` and every wire adapter collapses it (timeline→healthy, package rollup→healthy, AI summary→omit, topology node→gray). `neutral`-as-a-wire-value + all frontend rendering is **PR 2/2** — it can't be backend-only.

Net ~+1100 production lines: a shared documented package adds the abstraction the scattered copies never had. The consolidation reduces *duplication and sources of truth* (≈8 classifiers → 1, 4 rank maps → 1), not raw line count.

## Behavior changes (owned, intentional)

- **Topology: a crashlooping pod renders red, not green** — the headline fix (`getPodStatus` read only `pod.Status.Phase`). Visual-tested on a live kind cluster (crashloop / completed-job / scaled-to-zero / unschedulable fixtures).
- App/Add-ons rollup uses `health.Workload`: fixes the scale-down false-degraded (Spec vs Status replicas) and stops scaled-to-zero/suspended polluting rollups.
- AI context (`resourcecontext`): full classifier — a transiently-unready pod is healthy (not degraded); Deployments require availability (matching timeline + frontend); scaled-to-zero omits health.

## Exhaustive product review (200+ surface × kind × scenario combinations)

After the consolidation, every surface × resource-kind × scenario was reviewed (topology / resource tables / detail drawers / timeline / Issues / Applications / dashboard / MCP-AI / cross-surface), reading the handling code per scenario.

**Confirmed already-good (no change needed):**
- The **crashloop** state is now coherent across all 6 surfaces — the canonical classifier paying off where the old bug hurt most.
- **Issues panel** (pods + workloads/service/node): strong signal-to-noise — stable crashloop identity across the kubelet Waiting↔Running flap, 50-crashlooping-pods folds to one workload row, and benign states correctly suppressed (young Pending, recovered restarts, SchedulingGated, completed-Job-backed Services → "selector matches only completed pods" warning instead of a false critical).
- **Resource tables** (frontend, #1013-tuned): age-aware Pending/Terminating grace, fatal-container precedence, KEDA ScaledJob Idle-vs-NotReady ordering — all correct.
- **App rollup** is trustworthy in the safe direction: nothing genuinely broken reads green.
- The **neutral-collapse interim** (completed pod = sky table / gray topology / healthy timeline) is benign across surfaces — gray and sky both read "not a problem"; full unification is PR 2/2.

**Fixed by this PR** (gaps where a surface wasn't yet routed through the canonical classifier — all the same class: a benign/lifecycle state read green-for-broken or disagreed across surfaces):
- **Stuck-terminating pod (>10m) rendered green in topology** → now degraded, via a shared `health.IsStuckTerminating` (DRY with the timeline). Folded as a **floor** (`WorseOf`), so a crashlooping pod mid-deletion stays red.
- **CronJob topology node never migrated** — stale CronJob showed green, active showed amber → now via `health.Workload`. Also taught `cronJobVerdict` that an old never-scheduled CronJob is degraded (matching the Issues detector), not neutral.
- **Deployment/STS/DS/RS topology nodes** used a separate replica-ratio classifier → now `health.Workload` (Argo Rollout, an unstructured kind, keeps the ratio helper). Kills the Deployment ready-but-not-available green-vs-amber divergence.
- **Dashboard + MCP pod counters** bucketed node-lost (`Unknown`) pods as **Healthy** (the `LegacyString` collapse) → now switch on `Level` and count `unknown` as Warning, honoring the `PodUnknown→unknown` classification.
- **Timeline skipped Job/CronJob/PVC** (a failed Job never entered the Unhealthy filter) → now routed through `health.Workload`.

**Surfaced for a product decision (not changed here):** all-off app → green vs a distinct "Idle/Off" tier (PR 2/2); partially-degraded workload emitting `critical` (same tier as fully-down); a set of frontend detail-drawer over-alarms that contradict their own table badge (completed-Job-pod red containers, cordoned-node red banner, pending-PVC amber, terminating red banner at 60s) — all pre-existing TS-classifier contradictions belonging to PR 2/2.

## Testing

- Both modules `go build ./...`, `go vet`, and **all tests pass** (every `pkg/` package + `internal/k8s|server|mcp`). Edge-case + new-behavior coverage added (recovery guards, crashloop diagnosis ordering, node classification, stuck-terminating floor, never-scheduled CronJob, node-lost counters, timeline batch kinds).
- **Multi-cluster validation across 8 clusters** (GKE nonprod / management / prod / arlen, EKS, kind fixtures + crossplane-demo), API-first for breadth + Playwright for the notable states:
  - **Real clusters, healthy at scale** — GCP nonprod (93 deploy / cronjobs / jobs / pvc) and EKS (40 deploy): zero false positives, no crashes; a Job-class app correctly reads healthy.
  - **Real broken states** — GCP management: the app rollup surfaced **4 unhealthy + 1 degraded** apps with correct red "Down" / amber "Degraded" badges and 3 error pods on the dashboard. The neutral-collapse hid no real failure (the safe-direction property).
  - **Engineered states** — kind fixtures: a crashlooping pod renders **red** in topology + **Critical** in Issues + **Error** on the dashboard (coherent across surfaces); active CronJob → green, suspended → gray, completed Job → gray, scaled-to-zero → gray, unschedulable → amber with an actionable Issues message; all benign/lifecycle states correctly suppressed from the Issues panel.
- Cross-reviewed adversarially (Codex) across multiple passes, including a dedicated pass on the review fixes (which caught a floor-vs-ceiling downgrade — a stuck-terminating crashloop pod must stay red — before it shipped).

## Follow-up (PR 2/2)

Frontend + wire: backend emits the `Verdict` on list/detail/SSE; `getResourceStatus` consumes it / falls back to the TS classifier; `neutral` as its own tier (sky / Idle badge); the timeline span builder + detail-drawer severity fixes; topology renders neutral as sky.

---

> [!NOTE]
> **Medium Risk**
> Wide refactor of how health is computed across dashboard, MCP, topology, and issue detection—intentional semantic changes (counters, colors, rollups) could surprise operators even with heavy test coverage.
> 
> **Overview**
> Introduces **`pkg/health`** as the single backend classifier (`Level`, `Pod`/`Workload`/`Node`, shared `Rank`/`WorseOf`) and **removes** the duplicate logic that lived in `internal/k8s/health.go` plus per-surface helpers (`deploymentHealth`, `jobHealth`, phase-only topology pod status, etc.). Call sites in **problem detection**, **dashboard/MCP counters**, **timeline**, **topology**, **Applications/packages rollups**, and **resourcecontext** now call `health.*` directly, with small adapters mapping `neutral` onto existing 4-value wire vocabularies until a frontend follow-up.
> 
> **Notable behavior fixes:** topology and rollups use container-aware pod health (crashloops red, not green); display paths fold **unschedulable** and **stuck terminating** via `PodDisplayLevel` without downgrading real failures; dashboard/MCP treat **node-lost** (`Unknown`) pods as warning, not healthy; timeline classifies **Job/CronJob/PVC** through `health.Workload`.
> 
> Detector-only reason constants stay in `internal/k8s/reasons.go`; golden/recovery tests move with the package.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ec7a4e34950c31a7240626f6cabb332b75bed9fb. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: internal/k8s/detect.go [hunk 1]
```diff
@@ -355,25 +357,25 @@ func DetectProblems(cache *ResourceCache, namespace string) []Detection {
 				problems = append(problems, det)
 				continue
 			}
-			health := ClassifyPodHealth(pod, now)
+			healthStr := health.Pod(pod, now).LegacyString()
 			earlyProbeTargetProblem, hasEarlyProbeTargetProblem := activeProbeTargetProblem(pod, "")
-			if health == "healthy" && !hasEarlyProbeTargetProblem {
+			if healthStr == "healthy" && !hasEarlyProbeTargetProblem {
```
[the remark this answers is not in the document]
Comment (nadaverell) [reply internal/k8s/detect.go:362]: Intentional — skipping `LevelUnknown` (node-lost, `phase=Unknown`) pods here is noise reduction, not an oversight. When a node goes unreachable the **Node `NotReady` row** (`DetectNodeProblems`) is the single actionable signal; surfacing each stranded pod as its own problem row would just be noise that points at the same root cause.

These pods are **not** hidden, though: the REST + MCP dashboard counters now bucket `LevelUnknown` as **Warning** (via `health.Pod().Level`, not the legacy string) so they leave the healthy count — they just are not duplicated per-pod in the problem list. Deliberately leaving the detector as-is for that reason.
